### DISCUSSION TITLE
How to write properties on conformers (SDF)?

### DISCUSSION THREAD

**Discussion by jp-um** [rdkit/rdkit#3859] [Q&A]
Hi all, 

Long time no type.  Just following this up from a decade ago: https://sourceforge.net/p/rdkit/mailman/message/27547551/.  Greg had provided a nice and working hack, but I thought I'd revisit this.

I now noticed there is a `SetProp` on the `Conformer` class which I am pretty sure wasn't there when my hair was less grey.  However, I still can't get it to work (I've been trying on and off for 10 years, and now I am close to giving up :) ).

Minimal (non) working example:

```
from rdkit import Chem
from rdkit.Chem import AllChem

m = Chem.MolFromSmiles("c1ccccc1CNCC(=O)OC")
AllChem.EmbedMultipleConfs(m, 2) # no adding H etc, this is just a toy example

w = Chem.SDWriter("test.sdf")
confs = m.GetConformers()
for c in confs:
    c.SetProp("name", "dummy") # this is a specific value which is different per conformer
    w.write(m, confId=c.GetId())
w.close()
```

I must be doing something silly, but this code doesn't smell to me.  Note that if I write the property on `m`, this works.  I just want to confirm, that the properties on `Conformer` get ignored by the SDF writer (which I find surprising because the writer knows about the specific conformer with `confId`).

Many thanks, and have a great weekend,
JP Ebejer

**Comment by greglandrum** [comment]
Hi @jp-um,
Sorry for the slow reply; I didn't notice this question.

You can set properties on conformers but, as you've noticed, they are not written to the SDF. You still need to copy the properties from the Conformer to the Molecule when writing the SDF.

@bp-kelley maybe we could add an additional `extraProperties` argument to at least `SDWriter.write()` in order to allow the user to provide arbitrary extra properties for a molecule without having to go through the dance of actually calling `SetProp()` for each of them?

**Reply by bp-kelley** [reply]
Yeah, I expect we should come up with a sane default.  My preference is write mol prop data and conf data by default.  Confdata takes precedence over mol data.

**Reply by bp-kelley** [reply]
This may create an issue between reading and writing, but I’m okay with that.

**Reply by dww100** [reply]
Has anything happened with this idea?

**Reply by fpickard** [reply]
I am also interested in this proposal.
